### PR TITLE
🌱 Add description to m3cluster NoCloudProvider boolean field

### PR DESCRIPTION
**Pull request by Xenwar**
Add description for NoCloudProvider
The description is visible with kubectl explain m3mcluster.spec.NoCloudProvider

### CODE REVIEW

File: api/v1alpha4/metal3cluster_types.go [hunk 1]
```diff
@@ -32,7 +32,10 @@ const (
 type Metal3ClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
-	NoCloudProvider      bool        `json:"noCloudProvider,omitempty"`
+	// Determnes if the cluster will not be deployed with an external cloud provider.
+	// If set to true, CAPM3 will use node labels to set providerID on the kubernetes nodes.
+	// If set to false, providerID is set nodes by other entities and CAPM3 uses the value to the providerID on the m3m resource.
```
Comment (kashifest) [inline api/v1alpha4/metal3cluster_types.go:37]:
```suggestion
	// If set to false, providerID is set on nodes by other entities and CAPM3 uses the value of the providerID on the m3m resource.
```

File: api/v1alpha5/metal3cluster_types.go [hunk 2]
```diff
@@ -32,7 +32,10 @@ const (
 type Metal3ClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
-	NoCloudProvider      bool        `json:"noCloudProvider,omitempty"`
+	// Determnes if the cluster will not be deployed with an external cloud provider.
+	// If set to true, CAPM3 will use node labels to set providerID on the kubernetes nodes.
+	// If set to false, providerID is set nodes by other entities and CAPM3 uses the value to the providerID on the m3m resource.
```
Comment (kashifest) [inline api/v1alpha5/metal3cluster_types.go:37]: Same comment as above

File: api/v1beta1/metal3cluster_types.go [hunk 3]
```diff
@@ -32,6 +32,9 @@ const (
 type Metal3ClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
+	// Determnes if the cluster will not be deployed with an external cloud provider.
+	// If set to true, CAPM3 will use node labels to set providerID on the kubernetes nodes.
+	// If set to false, providerID is set nodes by other entities and CAPM3 uses the value to the providerID on the m3m resource.
```
Comment (kashifest) [inline api/v1beta1/metal3cluster_types.go:37]: same

**Review by kashifest — CHANGES_REQUESTED** [review]

File: api/v1alpha4/metal3cluster_types.go [hunk 4]
```diff
@@ -32,7 +32,10 @@ const (
 type Metal3ClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
-	NoCloudProvider      bool        `json:"noCloudProvider,omitempty"`
+	// Determnes if the cluster will not be deployed with an external cloud provider.
```
[the remark this answers is not in the document]
Comment (Xenwar) [reply api/v1alpha4/metal3cluster_types.go]: Thanks, done.

File: api/v1alpha4/metal3cluster_types.go (hunk 1 above)
Comment (Xenwar) [reply api/v1alpha4/metal3cluster_types.go]: Thanks, done.

File: api/v1alpha5/metal3cluster_types.go (hunk 2 above)
Comment (Xenwar) [reply api/v1alpha5/metal3cluster_types.go]: Thanks, done.

File: api/v1beta1/metal3cluster_types.go (hunk 3 above)
Comment (Xenwar) [reply api/v1beta1/metal3cluster_types.go]: Thanks, done.

**Comment by Xenwar** [comment]
//test-v1b1-integration

**Comment by furkatgofurov7** [comment]
/retitle 🌱 Add description to m3cluster NoCloudProvider boolean field
/lgtm

**Comment by kashifest** [comment]
/test-v1b1-integration
/approve
/lgtm
Adding lgtm since, it was added before by @furkatgofurov7

**Pull request merged**